apps/crn-server/test/data-providers/contentful/working-groups.data-provider.test.ts

```diff
@@ -0,0 +1,651 @@
+                    lastName: 'Doe',
+                    alumniSinceDate: null,
+                  },
+                },
+              ],
+            },
+          },
+        };
+
+        contentfulGraphqlClientMock.request.mockResolvedValueOnce(
+          contentfulGraphQLResponse,
+        );
+
+        const result = await workingGroupDataProvider.fetchById(id);
+
+        expect(result?.members).toHaveLength(1);
+        expect(result?.leaders).toHaveLength(0);
+      });
+    });
+  });
+
+  describe('Update method', () => {
+    test('Should update the working group with deliverables and delete previous ones', async () => {
+      const workingGroupId = 'working-group-id-1';
+
+      const workingGroupMock = getEntry({
+        deliverables: {
+          'en-US': [
+            {
+              sys: {
+                id: 'd0',
+              },
+              description: 'D0',
+              status: 'Complete' as DeliverableStatus,
+            },
+          ],
+        },
+      });
+      when(environmentMock.getEntry)
+        .calledWith(workingGroupId)
+        .mockResolvedValue(workingGroupMock);
+
+      const previousDeliverableMock = getEntry({});
+      previousDeliverableMock.isPublished = jest.fn(() => true);
+      when(environmentMock.getEntry)
+        .calledWith('d0')
+        .mockResolvedValue(previousDeliverableMock);
+
+      const deliverableCreatedMock = getEntry({});
+      environmentMock.createEntry.mockResolvedValue(deliverableCreatedMock);
+      deliverableCreatedMock.publish = jest
+        .fn()
+        .mockResolvedValueOnce(getEntry({}));
+
+      const workingGroupMockUpdated = getEntry({});
+      workingGroupMock.patch = jest
+        .fn()
+        .mockResolvedValueOnce(workingGroupMockUpdated);
+
+      await workingGroupDataProviderMock.update(workingGroupId, {
+        deliverables: [
+          {
+            description: 'D1',
+            status: 'Pending' as DeliverableStatus,
+          },
+        ],
+      });
+
+      expect(previousDeliverableMock.unpublish).toHaveBeenCalled();
+      expect(previousDeliverableMock.delete).toHaveBeenCalled();
+      expect(environmentMock.createEntry).toHaveBeenCalledWith(
+        'workingGroupDeliverables',
+        {
+          fields: {
+            description: { 'en-US': 'D1' },
+            status: { 'en-US': 'Pending' },
+          },
+        },
+      );
+      expect(deliverableCreatedMock.publish).toHaveBeenCalled();
+      expect(workingGroupMock.patch).toHaveBeenCalledWith([
+        {
+          op: 'replace',
+          path: '/fields/deliverables',
+          value: {
+            'en-US': [
+              {
+                sys: {
+                  id: 'entry-id',
+                  linkType: 'Entry',
+                  type: 'Link',
+                },
+              },
+            ],
+          },
+        },
+      ]);
+
+      // expect(workingGroupMockUpdated.publish).toHaveBeenCalled();
```

Comment:
i think you forgot something here

Reply:
done here [review: remove comment](https://github.com/yldio/asap-hub/pull/3110/commits/753df173434a994f498cad3804b12a9781aa7c81)